Tests/NetworkingMocks/URLProtocolMock.swift

```diff
@@ -0,0 +1,76 @@
+import Foundation
+
+class URLProtocolMock: URLProtocol {
+    // MARK: - Errors
+
+    private enum Errors: String, Error {
+        case noResponse = "no_response_to_match"
+        case noData = "data_is_nil"
+    }
+
+    // MARK: - Variables
+
+    static var requestResponses: [MockResponse] = []
+
+    // MARK: - Override URLProtocol Functions
+
+    override class func canInit(with request: URLRequest) -> Bool {
+        true
+    }
+
+    override class func canonicalRequest(for request: URLRequest) -> URLRequest {
+        request
+    }
+
+    override func startLoading() {
+        guard let client = client else { return }
+
+        // Here we try to match the request to the proper MockRequestResponse otherwise don't
+        guard
+            let firstIndex = Self.requestResponses
+                .firstIndex(where: { $0.canHandle(request: request) })
+        else {
+            client.urlProtocol(
+                self,
+                didFailWithError: Errors.noResponse
+            )
+            client.urlProtocolDidFinishLoading(self)
+            return
+        }
+        let requestResponse = Self.requestResponses.remove(at: firstIndex)
+        resolveRequest(with: requestResponse, client: client)
+    }
+
+    override func stopLoading() {
```

Comment:
Do we need to override all of these functions? If we can remove the unnecessary ones we should

Reply:
we need this as it's required for URLProtocol implementation :smile:

Reply:
Got it! This article helped me understand this test setup a bit more https://forums.raywenderlich.com/t/chapter-8-init-deprecated-in-ios-13/102050/8 for those also interested